### PR TITLE
Document `declaration-property-value-no-unknown` syntax reference

### LINKED ISSUE #8153: Document `declaration-property-value-no-unknown` syntax reference
### What minimal example or steps are needed to reproduce the bug?

```css
a { color: 0; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-property-value-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLpANhAE7woAMA3FgL4gA0IAZgJb4wBySAtnIjAB7cADmwB0YAM4T64CJhYBzBCAyYUKdCCIBXNhM2lV69Zthh8SIkhzM5AWiFEIQmERwBPOwDck+bTDtMCDttTABrIIB3bAQUHB0YLHVqGhlIeWYFADFiLmtlACsJORlYIWlEIw0QCQ82fGZMHANqixwYWs06JOra93rGnDt0xTtapEwoSygWzTaO5pBU6iA)

### Which Stylelint-related dependencies are you using?

[stylelint](https://stylelint.io/)@16.11.0
[stylelint-config-standard](https://www.npmjs.com/package/stylelint-config-standard)@36.0.1

### What did you expect to happen?

For the expected syntax to be included in the message, e.g.

```
"2:9-10 error Unexpected unknown value "0" for property "color", expected "<color>" (declaration-property-value-no-unknown)
```

### What actually happened?

```
"2:9-10 error Unexpected unknown value "0" for property "color" (declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

Perhaps include the expected syntax in problem message.

Some properties have quite long formal syntaxes, e.g. [`content`](https://csstree.github.io/docs/syntax/#Property:content):

```
normal | none | [ <content-replacement> | <content-list> ] [ / [ <string> | <counter> ]+ ]?
```

So, the messages may get too verbose.

A better solution may be to add a link to [CSSTree syntax reference](https://csstree.github.io/docs/syntax/#Property:content) in the [rule's README](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/) so that people can search for the property in question.

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

Closes #8153 

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/declaration-property-value-no-unknown/README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ It sometimes overlaps with:
 
 If duplicate problems are flagged, you can turn off the corresponding rule.
 
+For more details on CSS value definition syntax, refer to the [CSSTree Syntax Reference](https://csstree.github.io/docs/syntax/).
```
Comment (jeddy3) [inline lib/rules/declaration-property-value-no-unknown/README.md:27]: We tend to favour the active voice in our docs. How about we put the following after the 1st line and before the "This rule is only appropriate for CSS..." one:

> You can filter the [CSSTree Syntax Reference](https://csstree.github.io/docs/syntax/) to find out what value syntax is known for a property.

That'll keep the related bits of the extended description together.

**Review by jeddy3 — COMMENTED** [review]
@ryo-manba Thanks for getting the ball rolling with these PRs.

I've made a suggestion for the wording and position.

File: lib/rules/declaration-property-value-no-unknown/README.md (hunk 1 above)
Comment (ryo-manba) [reply lib/rules/declaration-property-value-no-unknown/README.md]: Good suggestion! I've applied it.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you.

**Pull request merged**